webcompat/static/js/lib/bugform.js

```diff
@@ -176,25 +176,23 @@ function BugForm() {
       $("[value=" + problemType[1] + "]").click();
     }
 
-    // If we have one or more details params,
+    // If we have a details params, JSON decode it and
     // add that to the end of the steps to reproduce textarea
-    var details = location.href.match(/details=([^&]*)/g);
-    if (details !== null) {
+    var details = location.href.match(/details=([^&]*)/);
+    if (details) {
       this.stepsToReproduceField.val(function(idx, value) {
         return (
           value +
           "\n" +
-          decodeURIComponent(
-            details
-              .map(function(item) {
-                return item.replace(/details=/, "");
-              })
-              .join("\n")
-              // The content of the details param may be encoded via
-              // application/x-www-form-urlencoded, so we need to change the
-              // + (SPACE) to %20 before decoding
-              .replace(/\+/g, "%20")
-          )
+          JSON.parse(
+            decodeURIComponent(
+              details[1]
+                // The content of the details param may be encoded via
+                // application/x-www-form-urlencoded, so we need to change the
+                // + (SPACE) to %20 before decoding
+                .replace(/\+/g, "%20")
+            )
+          ).join("\n")
```

Comment:
Here I see something related to return `\n`. But I don't see tests trying to handle this or maybe I don't understand how it is working. Should there be a test testing that the steps to reproduce field contains a list of things with hard returns, except of only one value as of now.

Reply:
We use `\n` just to make the information easier to digest in the textarea/final report. 
(Before, that was handled by Firefox). IMO, testing the formatting of the output isn't super important.

Reply:
I think I merged this too quickly! I'm gonna just send a JSON blob from firefox, then build up the entire content on the frontend -- and write tests for that. That'll mean less changes once we move to a hidden input as well (since the input wont know or care about `\n`'s or any particular format). Will send in another quick PR before deploying any of this.